frontend/src/components/filter/FilterContent.js

```diff
@@ -51,107 +114,120 @@ export default function FilterContent({
 
   const handleClickDialogClose = (prop, results) => {
     if (results) {
-      setCurrentFilters({ ...currentFilters, [prop]: results.map((x) => x.name) });
+      const updatedFilters = { ...currentFilters, [prop]: results.map((x) => x.name) };
+      setCurrentFilters(updatedFilters);
+      applyFilters(type, updatedFilters, isSmallScreen);
     }
+
     setOpen({ ...open, [prop]: false });
   };
 
   const handleValueChange = (key, newValue) => {
-    setCurrentFilters({ ...currentFilters, [key]: newValue });
+    const updatedFilters = { ...currentFilters, [key]: newValue };
+    applyFilters(type, updatedFilters, isSmallScreen);
+    setCurrentFilters(updatedFilters);
   };
 
   const handleApplyFilters = () => {
     applyFilters(type, currentFilters, isSmallScreen);
   };
 
   const handleUnselectFilter = (filterName, filterKey) => {
-    setCurrentFilters({
+    const updatedFilters = {
       ...currentFilters,
       [filterKey]: currentFilters[filterKey].filter((f) => f !== filterName),
-    });
-    if (selectedItems[filterKey])
+    };
+
+    // TODO(piper): verify this:
+    // When dismissing a selected filter chip, we also want to update the
+    // window state to reflect the currently active filters, and fetch
+    // the updated data from the server
+    applyFilters(type, updatedFilters, isSmallScreen);
```

Comment:
@piperchester to verify this:

```suggestion
    // When dismissing a selected filter chip, we also want to update the
    // window state to reflect the currently active filters, and fetch
    // the updated data from the server
    applyFilters(type, updatedFilters, isSmallScreen);
```